app/src/main/java/com/jerboa/MainActivity.kt

```diff
@@ -443,112 +380,68 @@ class MainActivity : AppCompatActivity() {
                     }
+
                     composable(
-                        route = "inbox",
+                        route = Route.INBOX,
                         deepLinks = DEFAULT_LEMMY_INSTANCES.map { instance ->
                             navDeepLink { uriPattern = "$instance/inbox" }
                         },
                     ) {
-                        if (account != null) {
-                            LaunchedEffect(Unit) {
-                                inboxViewModel.resetPage()
-                                inboxViewModel.getReplies(
-                                    GetReplies(
-                                        auth = account.jwt,
-                                    ),
-                                )
-                                inboxViewModel.getMentions(
-                                    GetPersonMentions(
-                                        auth = account.jwt,
-                                    ),
-                                )
-                                inboxViewModel.getMessages(
-                                    GetPrivateMessages(
-                                        auth = account.jwt,
-                                    ),
-                                )
-                            }
-                        }
-
                         InboxActivity(
                             navController = navController,
-                            inboxViewModel = inboxViewModel,
                             accountViewModel = accountViewModel,
-                            commentReplyViewModel = commentReplyViewModel,
                             siteViewModel = siteViewModel,
-                            privateMessageReplyViewModel = privateMessageReplyViewModel,
                         )
                     }
+
                     composable(
-                        route = "post/{id}",
+                        route = Route.POST,
                         deepLinks = DEFAULT_LEMMY_INSTANCES.map { instance ->
-                            navDeepLink { uriPattern = "$instance/post/{id}" }
+                            navDeepLink { uriPattern = "$instance/post/{${Route.PostArgs.ID}}" }
                         },
                         arguments = listOf(
-                            navArgument("id") {
-                                type = NavType.IntType
+                            navArgument(Route.PostArgs.ID) {
+                                type = Route.PostArgs.ID_TYPE
                             },
                         ),
                     ) {
-                        LaunchedEffect(Unit) {
-                            val postId = it.arguments?.getInt("id")!!
-                            postViewModel.initialize(id = Either.Left(postId))
-                            postViewModel.getData(account)
-                        }
-                        SwipeToNavigateBack(navController = navController) {
```

Comment:
#785 added a gesture to swipe to close comments. It would be great if that could be maintained.